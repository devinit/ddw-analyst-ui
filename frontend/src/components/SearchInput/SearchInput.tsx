import React, { FunctionComponent, KeyboardEvent, useEffect, useState } from 'react';
import { FormControl } from 'react-bootstrap';

interface SearchInputProps {
  onSearch?: (searchText: string) => void;
  placeholder?: string;
  className?: string;
  testid?: string;
  value?: string;
}

const SearchInput: FunctionComponent<SearchInputProps> = (props) => {
  const [searchText, setSearchText] = useState('');
  useEffect(() => setSearchText(props.value || ''), [props.value]);

  const onSearch = (event: KeyboardEvent<HTMLInputElement>) => {
    if (event.key === 'Enter' && props.onSearch) {
      event.preventDefault();
      event.stopPropagation();
      props.onSearch(searchText);
    }
  };

  return (
    <FormControl
      placeholder={props.placeholder}
      className={props.className}
      value={searchText}
      onChange={(event) => setSearchText(event.currentTarget.value)}
      onKeyDown={onSearch}
      data-testid={props.testid}
    />
  );
};

SearchInput.defaultProps = { testid: 'search-input', placeholder: 'Search ...' };

export { SearchInput };
