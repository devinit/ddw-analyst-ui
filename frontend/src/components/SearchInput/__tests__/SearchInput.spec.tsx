/**
 * @jest-environment jsdom
 */
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { SearchInput } from '../SearchInput';
import { fireEvent, render } from '@testing-library/react';

test('renders correctly with the default props', () => {
  const renderer = TestRenderer.create(
    <SearchInput testid="search-input" placeholder="Search ..." />,
  ).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('enter button triggers onChange function', () => {
  const onChange = jest.fn();
  const { getByTestId } = render(
    <SearchInput testid="search-input" placeholder="Search ..." onSearch={onChange} />,
  );
  fireEvent.keyDown(getByTestId('search-input'), { key: 'Enter', code: 13, charCode: 13 });

  expect(onChange).toHaveBeenCalled();
});
