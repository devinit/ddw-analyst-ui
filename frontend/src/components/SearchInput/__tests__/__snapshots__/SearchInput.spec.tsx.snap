// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with the default props 1`] = `
<input
  className="form-control"
  data-testid="search-input"
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Search ..."
  value=""
/>
`;
